test/e2e/tuning_options_test.go

```diff
@@ -320,3 +326,201 @@ func TestHAProxyTimeoutsRejection(t *testing.T) {
+			}
+
+			// Send a request through the router that includes the captured cookie, and verify that it's truncated as expected.
+
+			// Use the router image for the client pod since it includes curl.
+			routerDeployment := &appsv1.Deployment{}
+			routerDeploymentName := controller.RouterDeploymentName(ic)
+			if err := kclient.Get(context.TODO(), routerDeploymentName, routerDeployment); err != nil {
+				t.Fatalf("Failed to get router deployment for ingresscontroller %s: %v", icName.Name, err)
+			}
+
+			curlPodImage := ""
+			for _, container := range routerDeployment.Spec.Template.Spec.Containers {
+				if container.Name == "router" {
+					curlPodImage = container.Image
+					break
+				}
+			}
+			if curlPodImage == "" {
+				t.Fatalf("Failed to find container \"router\" in deployment %s", routerDeployment.Name)
+			}
+			curlPodName := types.NamespacedName{
+				Name:      names.SimpleNameGenerator.GenerateName("cookielen-curl-"),
+				Namespace: "default",
+			}
+
+			canaryRoute := &routev1.Route{}
+			canaryRouteName := controller.CanaryRouteName()
+			if err := kclient.Get(context.TODO(), canaryRouteName, canaryRoute); err != nil {
+				t.Fatalf("failed to get route %q: %v", canaryRouteName, err)
+			}
+			if canaryRoute.Status.Ingress == nil {
+				t.Fatalf("route %q has no hostname", canaryRouteName)
+			}
+			canaryRouteURL := ""
+			for _, canaryIngress := range canaryRoute.Status.Ingress {
+				if canaryIngress.RouterName == icName.Name {
+					canaryRouteURL = canaryIngress.Host
+					break
+				}
+			}
+			if canaryRouteURL == "" {
+				t.Fatalf("failed to find URL for route %q and ingress controller %q", canaryRouteName, icName)
+			}
+
+			icInternalService := &corev1.Service{}
+			icInternalServiceName := controller.InternalIngressControllerServiceName(ic)
+			if err := kclient.Get(context.TODO(), icInternalServiceName, icInternalService); err != nil {
+				t.Fatalf("failed to get service %q: %v", icInternalServiceName, err)
+			}
+
+			// Create a cookie that's one character over the max length of a captured cookie. Pad the cookie with a bunch of
+			// 'a's, and include an x at the end that should get truncated. The name of the cookie and the '=' are included in
+			// the max length, so account for that in the size of the padding.
+			padding := strings.Repeat("a", maxLength-len(cookieName)-1)
+			cookieString := fmt.Sprintf("%s=%sx", cookieName, padding)
+
+			curlArgs := []string{
+				"-k", "-L",
+				"-b", cookieString,
+				// --resolve used this way makes curl send the request to our ingress controller's internal service address as
+				// if it was directed to it by a load balancer. This ensures the request is handled by our ingress controller
+				// without having to wait extra time for a load balancer to be provisioned.
+				"--resolve", fmt.Sprintf("%s:443:%s", canaryRouteURL, icInternalService.Spec.ClusterIP),
```

Comment:
I don't think you need to do this at all. Can you explain?  It's better to use the established pattern mentioned in https://github.com/openshift/cluster-ingress-operator/pull/871/files#r1296550078, unless there is a need to do it differently.

Reply:
`TestUniqueIdHeader` does the same thing here: https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L3163 and there are a few other test cases that use this same behavior. It bypasses DNS and explicitly sends the request to the router pod(s) for a particular ingress controller, guaranteeing that the right router pod handles the connection, and removing any chance of slow DNS propagation causing test delays or failures.